File: GFG-DSA/Python/Mathematics/palindromeNumber.py
# palindrome number or not
n = int(input("Enter a number :"))
def palindrome(n):
    if (str(n) == str(n)[::-1]):
        return True
    else:
        return False
result = palindrome(n)

if(result):
    print(f"{n} is a palindrome number")
else:
    print(f"{n} is a not a palindrome number")


#solution 2 

# n = int(input("Enter a number :"))
# def pal(n):
#     rev = 0
#     temp = n
#     while(temp != 0):
#         lastdigit = temp % 10
#         rev = rev*10 + lastdigit
#         temp = temp//10
#     return rev == n

# result = pal(n)
# if(result):
#     print("Palindrome")
# else:
#     print("Not a Palindrome")

File: GFG-DSA/Java/Mathematics/palindromeNumber.java
import java.util.Scanner;

// palindrome number 
public class palindromeNumber {
    
    public static void main(String[] args) {
    
        Scanner scan = new Scanner(System.in);
        int num = scan.nextInt();
        System.out.println(isPalindrome(num));

        scan.close();
    }

    public static boolean isPalindrome(int num) {
        
        int temp = num; 
        int rev = 0; //321
        while (temp > 0) {
            rev = rev * 10 + temp % 10; //1
            temp /= 10;
        }
        if (num == rev) {
            return true;
        } else {
            return false;
        }

    }
}
